tests/test_calibrate.py

```diff
@@ -175,8 +179,144 @@ def test_calibrate_switches_to_simplified_correctly():
+            "tests/calibration_results/ihmh5_test",
         ]
     )
-    output_file = TEST_DIR / "ga_search_results/ihmh5_test/logbook.json"
+    output_file = TEST_DIR / "calibration_results/ihmh5_test/logbook.json"
     assert output_file.exists()
+
+
+def test_workflow_with_upgrade():
+    temp_output_dir = Path(tempfile.mkdtemp(prefix=f"calib_test_{uuid.uuid4().hex[:6]}_"))
+
+    # Create HPXML file for existing home
+    subprocess.run(
+        [
+            "openstudio",
+            "run",
+            "--workflow",
+            str(TEST_DIR / "data/ihmh3_existing_hpxml.osw"),
+            "--measures_only",
+        ],
+        capture_output=True,
+        check=True,
+    )
+    existing_hpxml_filepath = TEST_DIR / "data/uncalibrated_existing/ihmh3.xml"
+    assert existing_hpxml_filepath.exists()
+
+    # Run calibration
+    csv_bills_filepath = TEST_DIR / "data/ihmh3_existing_bills.csv"
+    config_filepath = TEST_DIR / "data/test_config.yaml"
+    cal_output_dir = TEST_DIR / "data/uncalibrated_existing/calibration_results"
+    app(
+        [
+            "calibrate",
+            str(existing_hpxml_filepath),
+            str(csv_bills_filepath),
+            "--output-dir",
+            str(cal_output_dir),
+            "--config-filepath",
+            str(config_filepath),
+            "--num-proc",
+            "8",
+        ]
+    )
+    cal_output_filepath = cal_output_dir / "logbook.json"
+    assert cal_output_filepath.exists()
+    cal_results = json.loads(cal_output_filepath.read_text())
+    cal_adjustments = cal_results["calibration_results"][-1]["best_individual"]
+
+    # FIXME: switch to success assert once
+    # https://github.com/NREL/OpenStudio-HPXML-Calibration/issues/91#issuecomment-3261124309
+    # is addressed
+    # FIXME: CI currently hits 20 gens even though we hit 9 gens locally
+    # assert cal_results["calibration_results"][-1]["gen"] < 20
+
+    # Create HPXML file for upgrade scenario (R-60 attic insulation)
+    subprocess.run(
+        [
+            "openstudio",
+            "run",
+            "--workflow",
+            str(TEST_DIR / "data/ihmh3_upgrade_hpxml.osw"),
+            "--measures_only",
+        ],
+        capture_output=True,
+        check=True,
+    )
+    upgrade_hpxml_filepath = TEST_DIR / "data/uncalibrated_upgrade/ihmh3.xml"
+    assert upgrade_hpxml_filepath.exists()
+
+    # Run upgrade scenario simulation
+    upgrade_run_dir = TEST_DIR / "data/uncalibrated_upgrade"
+    app(
+        [
+            "run-sim",
+            str(upgrade_hpxml_filepath),
+            "--output-dir",
+            str(upgrade_run_dir),
+            "--output-format",
+            "json",
+        ]
+    )
+    upgrade_json_filepath = upgrade_run_dir / "run/results_annual.json"
+    assert upgrade_json_filepath.exists()
+
+    # FUTURE: Move some methods outside of the Calibrate class so we
+    # can call them directly without having to create a cal object.
```

Comment:
If those methods are required for the full workflow, should we instead make CLI commands for them, so an external tool can use them?

Reply:
I generally don't think they will be needed by an external tool; but if we hear otherwise, we can do it at that time.

Reply:
I could be wrong, but I don't see them wanting to use these two methods. In any case, even if we did want to change it, it should go in a new PR. I want to get this merged in sooner.